python/lsst/daf/butler/gen2convert/walker.py

```diff
@@ -199,6 +240,16 @@ def scanRepo(self, repo):
                     log.debug("%s: found %s with %s", repo.root, dataset.datasetType.name, dataset.dataId)
                     repo.datasets[dataset.datasetType.name][filePath] = dataset
 
+                    try:
+                        calibRow = self._calibDict[(
+                            dataset.datasetType.name, dataset.dataId["calibDate"], dataset.dataId["ccd"])]
+                        if calibRow["filter"] is None or calibRow["filter"] == dataset.dataId["filter"]:
+                            dataset.dataId["valid_first"] = calibRow["valid_first"]
+                            dataset.dataId["valid_last"] = calibRow["valid_last"]
+                            print(calibRow)
```

Comment:
This `print` looks like a debugging relic.  It'd be nice to keep it, but via `log.debug` and a more specific message so we only get it when we turn the verbosity up.

Also, I *think* we might want to have an `else` clause here that does a `log.warn`, because it seems like it might be a problem if we find the dataset in the calib registry but don't understand its filter.  Is that right?

Reply:
This check is designed to allow datasets to either depend on the filter (flat/sky) or not (dark/bias) as needed.  This was a debug statement, so I've made it a log.debug() call.